pkg/kubelet/status/manager.go

```diff
@@ -332,12 +334,16 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 			m.deletePodStatus(uid)
 			return
 		}
+		if !m.needsUpdate(pod.UID, status) {
```

Comment:
Hmm..determining whether to send an update this late would mean that we need to always get the pod from the apiserver. I don't think we need to move this check.

If a mirror pod gets recreated, we will eventually see it from the watch, triggering a new call to `SetPodStatus`. At this time, if you check the `apiStatusVersions[newMirrorPod.UID]`, you would find nothing and send the update.